tests/test_xml_idenitifier.py

```diff
@@ -0,0 +1,63 @@
+import requests
+import httpretty
+from django.test import TestCase
+from bs4 import BeautifulSoup
+from publications.tasks import parse_html
+import xml.dom.minidom
+
+
+class SimpleTest(TestCase):
+
+    def parse_xml(dom):
+        collection = dom.documentElement
+        articles = collection.getElementsByTagName("dc:identifier") 
+        identifier_value = articles[0].firstChild.nodeValue
+        return identifier_value
+
+
+    def test_one(self):
+        xml_doc = """        
+        <metadata>
+            <oai_dc:dc xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/"
+                xmlns:dc="http://purl.org/dc/elements/1.1/"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/oai_dc/
+                http://www.openarchives.org/OAI/2.0/oai_dc.xsd">
+                <dc:title xml:lang="en-US">Test 1: One</dc:title>
+                <dc:creator>admin, admin</dc:creator>
+                <dc:description xml:lang="en-US">Test one</dc:description>
+                <dc:publisher xml:lang="en-US">Journal of Optimal Geolocations</dc:publisher>
+                <dc:date>2022-07-01</dc:date>
+                <dc:type>info:eu-repo/semantics/article</dc:type>
+                <dc:type>info:eu-repo/semantics/publishedVersion</dc:type>
+                <dc:type xml:lang="de-DE">Begutachter Beitrag</dc:type>
+                <dc:identifier>http://localhost:8330/index.php/opti-geo/article/view/1</dc:identifier>
+                <dc:source xml:lang="en-US">Journal of Optimal Geolocations; 2022</dc:source>
+                <dc:coverage xml:lang="en-US">Earth, Asia, Republic of Turkey</dc:coverage>
+                <dc:rights xml:lang="de-DE">Copyright (c) 2022 Journal of Optimal Geolocations</dc:rights>
+            </oai_dc:dc>
+        </metadata>              
+        """        
+        dom = xml.dom.minidom.parseString(xml_doc)
+        url_link = SimpleTest.parse_xml(dom)
+
+        httpretty.enable(verbose=True, allow_net_connect=False)  # enable HTTPretty so that it will monkey patch the socket module
+        httpretty.register_uri(httpretty.GET, "http://localhost:8330/index.php/opti-geo/article/view/1",
+                            body="""
```

Comment:
You could also load the content from the HTML files, `body = readFile(..)`.

https://www.askpython.com/python/examples/read-file-as-string-in-python